src/encord_active/analysis/types.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import uuid
+from dataclasses import dataclass
+from typing import Annotated, Dict, NamedTuple, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from encord_active.db.models import AnnotationType
+
+
+class Point(NamedTuple):
+    x: int
+    y: int
+
+
+@dataclass(frozen=True)
+class AnnotationMetadata:
+    feature_hash: str
+    annotation_type: AnnotationType
+    annotation_email: str
+    annotation_manual: bool
+    annotation_confidence: float
+    points: Optional[PointTensor]
+    mask: Optional[MaskTensor]
+    bounding_box: Optional[BoundingBoxTensor]
+
+
+@dataclass(frozen=True)
+class NearestNeighbors:
+    metric_deps: list[MetricDependencies]
+    metric_keys: list[Union[Tuple[uuid.UUID, int], Tuple[uuid.UUID, int, str]]]
+    similarities: list[float]
```

Comment:
What if we change `similarities` to `distances`, which is more self-explanatory?